01-CodeReviewEx/Shop-Ex/Program.cs

```diff
@@ -12,9 +12,33 @@ static void Main(string[] args)
         {
             
          
+         Account[] myAccounts = new Account[10];
+            Account a = new Account();
+            a.principal = 1.0;
+            a.rate = 0.5;
+            myAccounts[0] = a;
+            a = new Account() { principal = 1.5, rate = 0.75 };
+            
+            double fee = calculateFee(myAccounts);
+            Console.WriteLine(fee);
+         
+        }
+
+        public static double calculateFee(Account[] accounts)
+        {
+            double totalFee = 0.0;
+            Account account;
+            for (int i = 0; i < accounts.Length; i++)
+            {
+                account = accounts[i];
+                if (account.accountType == Account.PREMIUM || account.accountType == Account.PREMIUM_PLUS) totalFee += .0125 * ( // 1.25% broker's fee
+              account.principal * Math.pow(account.rate, (account.daysActive / 365.25)) - account.principal); // interest-principal
```

Comment:
I would use similar approach for 365.25 days of the year referenced here (even though 365.25 is self-descriptive in a way). Also - what about leap years, do they make a difference from the logic's perspective?